### PR TITLE
Update examples and docs for resource.name contains

### PR DESCRIPTION
**Pull request by JoshuaWilkes**
Update resource.name comparisons to match the cedar docs https://docs.cedarpolicy.com/policies/syntax-operators.html

### CODE REVIEW

**Review by ckluy31 — CHANGES_REQUESTED** [review]

File: examples/resources/commonfate_entra_group_availabilities/commonfate_entra_group_selector/resource.tf [hunk 1]
```diff
@@ -11,6 +11,6 @@ resource "commonfate_entra_group_selector" "name_contains" {
   name      = "Select Production Entra Groups"
   tenant_id = "00551d20-529d-478f-b1cb-fb04a2653e97"
   when      = <<EOF
-  resource.name contains "production"
+  resource.name like "*production*
```
Comment (JoshuaWilkes) [inline examples/resources/commonfate_entra_group_availabilities/commonfate_entra_group_selector/resource.tf:14]:
```suggestion
  resource.name like "*production*"
```

File: examples/resources/commonfate_entra_group_selector/resource.tf [hunk 2]
```diff
@@ -11,6 +11,6 @@ resource "commonfate_entra_group_selector" "name_contains" {
   name      = "Select Production Entra Groups"
   tenant_id = "00551d20-529d-478f-b1cb-fb04a2653e97"
   when      = <<EOF
-  resource.name contains "production"
+  resource.name like "*production*
```
Comment (JoshuaWilkes) [inline examples/resources/commonfate_entra_group_selector/resource.tf:14]:
```suggestion
  resource.name like "*production*"
```

File: examples/resources/commonfate_okta_group_selector/resource.tf [hunk 3]
```diff
@@ -11,6 +11,6 @@ resource "commonfate_okta_group_selector" "name_contains" {
   name            = "Select Production Okta Groups"
   organization_id = "dev-12345678"
   when            = <<EOF
-  resource.name contains "production"
+  resource.name like "*production*
```
Comment (JoshuaWilkes) [inline examples/resources/commonfate_okta_group_selector/resource.tf:14]:
```suggestion
  resource.name like "*production*"
```

File: internal/auth0/resource_auth0_organization_selector.go [hunk 4]
```diff
@@ -95,7 +95,7 @@ func (r *Auth0OrganizationSelectorResource) Schema(ctx context.Context, req reso
 			},
 
 			"when": schema.StringAttribute{
-				MarkdownDescription: "A Cedar expression with the criteria to match groups on, e.g: `resource.name contains \"production\"",
+				MarkdownDescription: "A Cedar expression with the criteria to match groups on, e.g: ``resource.name like \"*production*\"`",
```
Comment (JoshuaWilkes) [inline internal/auth0/resource_auth0_organization_selector.go:98]:
```suggestion
				MarkdownDescription: "A Cedar expression with the criteria to match groups on, e.g: `resource.name like \"*production*\"`",
```

File: docs/resources/entra_group_selector.md [hunk 5]
```diff
@@ -25,7 +25,7 @@ resource "commonfate_entra_group_selector" "name_contains" {
   name      = "Select Production Entra Groups"
   tenant_id = "00551d20-529d-478f-b1cb-fb04a2653e97"
   when      = <<EOF
-  resource.name contains "production"
+  resource.name like "*production*"
```
[the remark this answers is not in the document]
Comment (JoshuaWilkes) [reply docs/resources/entra_group_selector.md:28]: This one should not have the `

**Review by ckluy31 — APPROVED** [review]

**Pull request merged**